tests/test_views_objectversion.py

```diff
@@ -172,169 +183,186 @@ def test_get_unreadable_file(client, headers, bucket, objects, db,
     obj.file.readable = False
     db.session.commit()
 
-    resp = client.get(url_for(
-        'invenio_files_rest.object_api',
-        bucket_id=bucket.id,
-        key=obj.key,
-    ))
+    resp = client.get(
+        url_for(
+            "invenio_files_rest.object_api",
+            bucket_id=bucket.id,
+            key=obj.key,
+        )
+    )
     assert resp.status_code == 503
 
 
-def test_get_versions(client, headers, bucket, versions, permissions):
-    """Test object version getting."""
-    cases = [
+@pytest.mark.parametrize(
+    "user, expected",
+    [
         (None, 404),
-        ('auth', 404),
-        ('objects', 403),
-        ('bucket', 403),
-        ('location', 200),
-    ]
-
-    for user, expected in cases:
-        login_user(client, permissions[user])
+        ("auth", 404),
+        ("objects", 403),
+        ("bucket", 403),
+        ("location", 200),
+    ],
+)
+def test_get_versions(client, headers, bucket, versions, permissions, user, expected):
+    """Test object version getting."""
+    login_user(client, permissions[user])
 
-        for obj in versions:
-            if obj.is_head is True:
-                continue
-            resp = client.get(
-                url_for(
-                    'invenio_files_rest.object_api',
-                    bucket_id=bucket.id, key=obj.key, ),
-                query_string=dict(versionId=obj.version_id)
-            )
-            assert resp.status_code == expected
+    for obj in versions:
+        if obj.is_head is True:
+            continue
+        resp = client.get(
+            url_for(
+                "invenio_files_rest.object_api",
+                bucket_id=bucket.id,
+                key=obj.key,
+            ),
+            query_string=dict(versionId=obj.version_id),
+        )
+        assert resp.status_code == expected
 
-            if resp.status_code == 200:
-                # Strips prefix 'md5:' from checksum value.
-                assert resp.content_md5 == obj.file.checksum[4:]
-                assert resp.get_etag()[0] == obj.file.checksum
+        if resp.status_code == 200:
+            # Strips prefix 'md5:' from checksum value.
+            assert resp.content_md5 == obj.file.checksum[4:]
+            assert resp.get_etag()[0] == obj.file.checksum
 
 
-def test_get_versions_invalid(client, headers, bucket, objects, permissions):
-    """Test object version getting."""
-    cases = [
+@pytest.mark.parametrize(
+    "user",
+    [
         None,
-        'auth',
-        'objects',
-        'bucket',
-        'location',
-    ]
-
+        "auth",
+        "objects",
+        "bucket",
+        "location",
+    ],
+)
+def test_get_versions_invalid(client, headers, bucket, objects, permissions, user):
+    """Test object version getting."""
     versions = [
```

Comment:
could this also be replaced by `parametrize`?
this would make its use a bit more consistent

Reply:
I've thought about it, but i don't know how to use the parametrize feature as a double for loop.

Reply:
fair enough